checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -575,8 +591,27 @@ private void computeFieldAccessType(AnnotatedTypeMirror type,
                     .asType();
             boolean isInitializedForFrame = isInitializedForFrame(receiverType, fieldDeclarationType);
             if (isInitializedForFrame) {
+                // receiver is initialized for this frame.
+                // change the type of the field to @UnknownInitialization so that
+                // anything can be assigned to this field.
                 type.replaceAnnotation(qualHierarchy.getTopAnnotation(UNCLASSIFIED));
+
+            } else if(computingAnnotatedTypeMirrorOfLHS) {
+                // receiver is not initialized for this frame and the type of a lhs is being computed
+                // so replace all annotations with top expect for the field invariant annotation hierarchy
+                // (Nullness, for example)
+                AnnotationMirror childTypeSystemAnnotation = type.getAnnotationInHierarchy(getFieldInvariantAnnotation());
```

Comment:
Can you explain the logic of this code? By doing the type.getAIH(getFieltInvariantA) you get whatever qualifier from the child hierarchy the type contains. Then you only replace annotations that are not in that hierarchy. So for Nullness, you don't replace the annotation with the top annotation. I would have expected from the documentation that you replace everything other than the field invariant annotation with top.